### PR TITLE
Minor improvement for APGD

### PR DESCRIPTION
**Pull request by CNOCycle**
This PR is a minor improvement.

1. Remove function `lp_norm`, which has been defined in file `other_utils`
2. Initialize vars and run assertion in constructor
3. Refine function `normalize`, but I prefer the following implementation
```python
de = t.view(-1, *([1] * self.ndims))
ans = torch.where(de < 1e-12, x / (de + 1e-12), x / de)
```

Also, I noticed that function `check_shape` has been defined in APGD and FAB, but `check_shape` dose not be called in APGD. Similarly, some util functions are re-defined in Square attack. I would suggest that those functions should be moved to `other_utils`. Second, var `adv_all` at end of Square attack is not defined.

### CODE REVIEW

**Comment by fra31** [comment]
Hi,

thanks for the improvements! I agree with you other suggestions too, I'll try to integrate them as soon as possible.
